### PR TITLE
netsurf: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Description of changes

To a more modern style.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Please use the "mark as ready for review" button only when you fullfill the CONTRIBUTING guide conditions.
This PR has no description, no information in the title.

**Comment by AndersonTorres** [comment]
> Please use the "mark as ready for review" button only when you fullfill the CONTRIBUTING guide conditions. This PR has no description, no information in the title.

The code (and the individual commits) were ready for review. If your only complain is about this, you could just ask.

**Comment by RaitoBezarius** [comment]
> > Please use the "mark as ready for review" button only when you fullfill the CONTRIBUTING guide conditions. This PR has no description, no information in the title.
> 
> The code (and the individual commits) were ready for review. If your only complain is about this, you could just ask.
> 
> Indeed, it would be part of the code review itself! The button says "mark as ready for **review**", not "mark as perfect for _merge_".

No, no, people are monitoring PRs and are taking time to review, please respect their time and mark it as ready when it has sufficient information, do not make reviewers ask you for information as much as possible, otherwise we just cannot deal with the sheer amount of reviews. Thank you for your understanding.

**Comment by AndersonTorres** [comment]
Is there any info that needs to be added?

File: pkgs/applications/networking/browsers/netsurf/buildsystem.nix [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 
-  meta = with lib; {
+  meta = {
     homepage = "https://www.netsurf-browser.org/";
     description = "NetSurf browser shared build system";
-    license = licenses.mit;
-    maintainers = [ maintainers.vrthra maintainers.AndersonTorres ];
-    platforms = platforms.unix;
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ samueldr vrthra AndersonTorres ];
```
Comment (samueldr) [inline pkgs/applications/networking/browsers/netsurf/buildsystem.nix:23]:
```suggestion
    maintainers = with lib.maintainers; [ vrthra AndersonTorres ];
```

Sorry, I haven't been doing a job (good or bad) maintaining this.

And thank you for doing a refresh!

**Pull request merged**